go-controller/pkg/clustermanager/egressservice/egressservice_cluster.go

```diff
@@ -479,7 +492,27 @@ func (c *Controller) syncEgressService(key string) error {
 		return c.clearServiceResourcesAndRequeue(key, state)
 	}
 
-	// At this point both the EgressService resource and the Service != nil
+	// At this point the Service has at least one ingress IP and its EgressService != nil.
+	// We check if it its sourceIPBy != LoadBalancerIP to determine if we need to clean its resources,
+	// set host=ALL and stop processing or not.
+	if es.Spec.SourceIPBy != egressserviceapi.SourceIPLoadBalancer {
+		if state == nil {
+			// The service does not need SNAT LRPs and was not an allocated egress service.
+			// We delete it from the unallocated service cache just in case and set its host
+			// to "ALL".
+			delete(c.unallocatedServices, key)
+			return c.setEgressServiceHost(namespace, name, noSNATHost)
+		}
+
+		err := c.clearServiceResourcesAndRequeue(key, state)
```

Comment:
I modified the function to accept a new "serviceHost" enum (that represents hosts with special meanings) to set on the status instead of defaulting to "", is that approach fine?